docs/ContribOperators.md

```diff
@@ -2454,6 +2455,49 @@ This version of the operator has been available since version 1 of the 'com.micr
 </dl>
 
 
+### <a name="com.microsoft.SparseToDenseMatMul"></a><a name="com.microsoft.sparsetodensematmul">**com.microsoft.SparseToDenseMatMul**</a>
+
+#### Version
+
+This version of the operator has been available since version 1 of the 'com.microsoft' operator set.
+
+#### Attributes
+
+<dl>
+<dt><tt>alpha</tt> : float</dt>
+<dd>Scalar multiplier for the product of the input tensors.</dd>
+<dt><tt>transA</tt> : int</dt>
+<dd>Whether A should be transposed on the last two dimensions before doing multiplication</dd>
+<dt><tt>transB</tt> : int</dt>
+<dd>Whether B should be transposed on the last two dimensions before doing multiplication</dd>
+</dl>
+
+#### Inputs
+
+<dl>
+<dt><tt>A</tt> : T</dt>
+<dd>2-dimensional sparse matrix A. Either COO or CSR format</dd>
```

Comment:
Shall we have design some constraint (like type constraint) for storage format that the operator supported? In this way, we can have some logic to implicitly convert the storage format to fit an operator (like add a format "Cast" node in execution plan).

Instead of type constraint like sparse_tensor(float), tensor(float). We can have some constraints like tensor(float), tensor_format(dense, coo, csr).

Reply:
This was discussed and it not clear how to implement it in the view of the possible number of permutations. The type registration along would take many types.

Reply:
Currently, we are thinking of adding this to KernelDef so it can be enforced and a selection can be made between potentially different kernels.